plugins:
    - {name: obstacle_layer, type: "costmap_2d::ObstacleLayer"}
    - {name: inflation_layer, type: "costmap_2d::InflationLayer"}
    - {name: orne_flat_layer, type: "orne_flat_layer_namespace::OrneFlatLayer"}
    - {name: scan_obstacle, type: "costmap_2d::VoxelLayer"}
    - {name: scan_inflation, type: "costmap_2d::InflationLayer"}
    # - {name: point_cloud_obstacle, type: "costmap_2d::VoxelLayer"}
    # - {name: point_cloud_inflation, type: "costmap_2d::InflationLayer"}
    #
map_type: voxel
origin_z: 0.0
z_resolution: 0.2
z_voxels: 10
unknown_threshold: 10
mark_threshold: 0

obstacle__range: 2.5
max_obstacle_height: 2.0
raytrace_range: 3.0

obstacle_layer:
    observation_sources: point_cloud
    point_cloud: {data_type: PointCloud2, topic: /filtered_cloud, marking: true, clearing: true}

scan_obstacle:
    observation_sources: scan obstacle_cloud
    scan: {data_type: LaserScan, topic: /scan, marking: true, clearing: true, min_obstacle_height: 0.3, max_obstacle_height: 1.0}
    obstacle_cloud: {data_type: PointCloud2, topic: /obstacle_cloud, marking: true, clearing: true}

# point_cloud_obstacle:
#     observation_sources: obstacle_cloud
#     obstacle_cloud: {data_type: PointCloud2, topic: /obstacle_cloud, marking: true, clearing: true}
